modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java

```diff
@@ -601,9 +599,9 @@ public boolean test(String profile, InetSocketAddress peerAddress) {
         ) {
             transport.start();
             int nRetries = randomIntBetween(7, 9);
-            for (int i = 0; i < nRetries; i++) {
-                acceptChannel.set(randomBoolean());
-                try (Netty4HttpClient client = new Netty4HttpClient()) {
+            try (Netty4HttpClient client = new Netty4HttpClient()) {
```

Comment:
Alternatively, I think it's "better" to reuse the client. If there is context pollution, it will first show up across different requests from the same client, before leaking across clients (because clients are statically assigned to a single thread).